k8s/upgrade/src/plugin/upgrade.rs

```diff
@@ -833,3 +930,43 @@ pub(crate) async fn get_source_version(ns: &str) -> error::Result<String> {
         .to_string();
     Ok(value.to_string())
 }
+
+/// Parse set-file and create config map data.
+pub(crate) async fn create_config_map_data(
+    upgrade_args: &UpgradeArgs,
+) -> error::Result<BTreeMap<String, String>> {
+    let mut upgrade_map = BTreeMap::new();
+    for file in &upgrade_args.set_file {
+        let data: Vec<_> = file.split('=').collect();
+        let cm_values =
+            fs::read_to_string(data[1]).context(error::ReadFromFile { filepath: data[1] })?;
+        // create config map key with key of set-file
+        // Example : -- set-file jaeger-operator.tolerations=/root/tolerations.yaml
+        // Key:  jaeger-operator.tolerations
+        // this is acts as a name of file under the mounted path inside pod.
+        let cm_key = data[0].to_string();
```

Comment:
I think we shouldn't have 1 file per key. This creates duplicate files when using the same file for several helm values (e.g. setting tolerations for multiple components). 

IMO, we should have 1 configmap entry per file. If the file names are the same, then the configmap key could be appended with random suffixes, or a '-' separated version of their absolute paths.

Reply:
The present implementation  does the below 
--set-args  jaeger-operator.tolerations=/home/ashish/tolerations.yaml
The key is `jaeger-operator.tolerations` , name of file is `jaeger-operator.tolerations` and the contents of file will be written in file`/upgrade-config-map/jaeger-operator.tolerations` inside pod.

Problem with having 1 configmap entry per file

Example `--set-file jaeger-operator.tolerations=/root/tolerations.yaml --set-file ha.node.tolerations=/root/temp/random/tolerations.yaml`

In this case if we work with file name as value then both the keys `jaeger-operator.tolerations` and `ha.node.tolerations` will have value as tolerations.yaml ( the file name) ; even though the files are different .

To solve this we need to keep a track of the files names and check if its same , in case its same then we add a random number to file name to make them unique.

All these , i.e. finding a file name. checking the file to the previous file name and then appending random number to file name adds to code debt. 

The solution to this problem is to create 1 config map per key. Yes agreed it will increase the number of config map but will save us from doing al the above hacks. 

There can be multiple cm for upgrade, its the same as the users are passing multiple set args in command. And these cm will be deleted once the upgrade resources are deleted. 

@tiagolobocastro  need your input here.

Reply:
I think I have covered this case in my comment above. file absolute paths are the same, we can avoid using the filename as the configmap key, and instead use the absolute path separated by '-' as the configmap key (which eventually turn into a file inside the container). It is impossible that two separate files will have the same absolute path.

Reply:
Can we not just use a tmp file name for the configmap? Usually tmp file names are reasonably unique and chances of collision is next to none. https://man7.org/linux/man-pages/man3/tmpfile.3.html

Reply:
Yes, the same logic applies here. However because we're not creating the file ourselves, and instead we're creating the configmap entry. We'd have to use a random filename/filename suffix instead. Yes, this will work.

Reply:
Just wanted to ask here , whats the issue with having one file per key?

Reply:
> Just wanted to ask here , whats the issue with having one file per key?

Yes, and the filename can be randomly generated using tmp file logic.

Reply:
> Just wanted to ask here , whats the issue with having one file per key?
@sinhaashish -- It is a copy of the same file...

> Yes, and the filename can be randomly generated using tmp file logic.
With <one-file-per-key> (this is not one-key-per-file), the file names are unique, as all helm values keys are unique. But, if multiple keys make use of data from the same file, then the file is copied for each of these keys. Because, one file per key, even if the file is the same file. The unique file name problem doesn't show up with one file per key.